source/common/http/http1/codec_impl.cc

```diff
@@ -101,20 +104,12 @@ StreamEncoderImpl::StreamEncoderImpl(ConnectionImpl& connection,
   }
 }
 
-void StreamEncoderImpl::encodeHeader(const char* key, uint32_t key_size, const char* value,
-                                     uint32_t value_size) {
-
-  ASSERT(key_size > 0);
-  const uint64_t old_buffer_length = connection_.buffer().length();
-  connection_.copyToBuffer(key, key_size);
-  connection_.addCharToBuffer(':');
-  connection_.addCharToBuffer(' ');
-  connection_.copyToBuffer(value, value_size);
-  connection_.addToBuffer(CRLF);
-  bytes_meter_->addHeaderBytesSent(connection_.buffer().length() - old_buffer_length);
-}
 void StreamEncoderImpl::encodeHeader(absl::string_view key, absl::string_view value) {
-  this->encodeHeader(key.data(), key.size(), value.data(), value.size());
+  ASSERT(!key.empty());
+
+  const uint64_t header_size = connection_.buffer().addFragments(key, COLON, SPACE, value, CRLF);
```

Comment:
Please introduce a new variable COLONSPACE = ": "

Reply:
get it.